cli/src/pcluster/resources/custom_resources/custom_resources_code/cleanup_resources.py

```diff
@@ -117,12 +119,39 @@ def _terminate_cluster_nodes(event):
                         logger.error("Failed when terminating instances with error %s", e)
                         completed_successfully = False
                         continue
-        logger.info("Compute fleet clean-up: COMPLETED. Instances are either in shutting-down or terminated state")
+            logger.info("Sleeping for 10 seconds to allow all instances to initiate shut-down")
+            time.sleep(10)
+
+        while _has_shuttingdown_instances(stack_name):
+            # This logic prevents Lambda function from timing out if instance termination exceeds 15 minutes
+            # TODO: This approach may cause potential cluster deletion failure when PlacementGroups are enabled
+            # TODO: and instance termination time exceeds 15 minutes simultaneously.
+            # TODO: Resolve the above potential failure.
+            if time.time() - start_time > max_time:
+                return
+            logger.info("Waiting for all nodes to shut-down...")
+            time.sleep(10)
+
+        # Sleep for 30 more seconds to give PlacementGroups the time to update
+        time.sleep(30)
+
+        logger.info("Compute fleet clean-up: COMPLETED")
```

Comment:
Cleanup is not the right term. Here we are waiting for the compute nodes to terminate, right? If so, make the log more specific